tcbuilder/backend/build.py

```diff
@@ -317,6 +318,12 @@ def make_feed_url(feed_props):
         raise InvalidDataError(
             f"Don't know how to handle a major version of {version_major}")
 
+    if params["variant"] is None:
+        if MAJOR_TO_YOCTO_MAP[version_major] in ("dunfell-5.x.y", "kirkstone-6.x.y"):
+            params["variant"] = DEFAULT_LEGACY_IMAGE_VARIANT
+        else:
+            params["variant"] = DEFAULT_IMAGE_VARIANT
```

Comment:
Instead of an `else` here should we explicitly check if the `version_major` is even scarthgap based or not?

Reply:
The `if version_major` above should prevent any other major from being entered in which case I think the condition here is okay considering that we probably don't want to change the name of the variant any time soon. When another major of TOS is available we'd add it to `MAJOR_TO_YOCTO_MAP` but the code here wouldn't need to be changed.

Reply:
> The if version_major above should prevent any other major from being entered in which case I think the condition here is okay considering that we probably don't want to change the name of the variant any time soon. When another major of TOS is available we'd add it to MAJOR_TO_YOCTO_MAP but the code here wouldn't need to be changed.

Exactly, that's why I added the variant check after the `if version_major` check. And in `MAJOR_TO_YOCTO_MAP` 5 is the lowest version defined, so the `else` statement will only be reached with major version 7 (and above when they're eventually defined in `MAJOR_TO_YOCTO_MAP`).